java/ql/src/semmle/code/java/frameworks/Flexjson.qll

```diff
@@ -0,0 +1,51 @@
+/**
+ * Provides classes for working with the Flexjson framework.
+ */
+
+import java
+
+/** The class `flexjson.JSONDeserializer`. */
+class FlexjsonDeserializer extends RefType {
+  FlexjsonDeserializer() { this.hasQualifiedName("flexjson", "JSONDeserializer") }
+}
+
+/** The class `flexjson.JSONSerializer`. */
+class FlexjsonSerializer extends RefType {
+  FlexjsonSerializer() { this.hasQualifiedName("flexjson", "JSONSerializer") }
+}
+
+/** The deserialization method `deserialize`. */
+class FlexjsonDeserializeMethod extends Method {
+  FlexjsonDeserializeMethod() {
+    (
+      this.getDeclaringType().(ParameterizedType).getGenericType() instanceof FlexjsonDeserializer or
+      this.getDeclaringType().getSourceDeclaration().getASourceSupertype*() instanceof
+        FlexjsonDeserializer
+    ) and
+    this.getName().matches("deserialize%")
+  }
+}
+
+/** The serialization method `serialize`. */
+class FlexjsonSerializeMethod extends Method {
+  FlexjsonSerializeMethod() {
+    (
+      this.getDeclaringType().(ParameterizedType).getGenericType() instanceof FlexjsonSerializer or
+      this.getDeclaringType().getSourceDeclaration().getASourceSupertype*() instanceof
+        FlexjsonSerializer
+    ) and
+    this.hasName(["serialize", "deepSerialize"])
+  }
+}
+
+/** The method `use` to configure allowed class type. */
+class DeserializerUseMethod extends Method {
+  DeserializerUseMethod() {
+    (
+      this.getDeclaringType().(ParameterizedType).getGenericType() instanceof FlexjsonDeserializer or
```

Comment:
This might be redundant; `getSourceDeclaration()` (in the line below) should cover parameterized types, see its [documentation](https://codeql.github.com/codeql-standard-libraries/java/semmle/code/java/Type.qll/predicate.Type$RefType$getSourceDeclaration.0.html).

Reply:
Thanks @Marcono1234. I've made the change and did a force push.